deepchem/metalearning/maml.py

```diff
@@ -0,0 +1,226 @@
+"""Model-Agnostic Meta-Learning (MAML) algorithm for low data learning."""
+
+from deepchem.models.tensorgraph.layers import Layer
+from deepchem.models.tensorgraph.optimizers import Adam
+import os
+import shutil
+import tempfile
+import tensorflow as tf
+import time
+
+class MetaLearner(object):
+  """Model and data to which the MAML algorithm can be applied.
+  
+  To use MAML, create a subclass of this defining the learning problem to solve.
+  It consists of a model that can be trained to perform many different tasks, and
+  data for training it on a large (possibly infinite) set of different tasks.
+  """
+
+  def __init__(self):
```

Comment:
I thought I'd deleted that!  Not sure what happened.  Right, the parent class isn't supposed to have a constructor.